File: Java/Search Insert Position.java
// 0ms, 100.00%; 41MB, 5.41%
class Solution {
    public int searchInsert(int[] nums, int target) {
        /* find first element that >= target */
        int left = 0, right = nums.length;
        while (left < right) {
            int mid = left + (right - left) / 2;
            if (nums[mid] < target) {
                left = mid + 1;
            } else {
                right = mid;
            }
        }
        return left;
    }
}

File: Go/Search Insert Position.go
// 4ms, 84.16%; 3.1MB, 100.00%
func searchInsert(nums []int, target int) int {
	i, j := 0, len(nums)-1
	if nums[j] < target {
		return len(nums)
	}
	for i < j {
		mid := i + (j-i)/2
		if nums[mid] < target {
			i = mid + 1
		} else {
			j = mid
		}
	}
	return i
}
